namespaces/referencing-namespace.md

```diff
@@ -0,0 +1,194 @@
+# Referencing Namespace
+
+This namespace defines elements and attributes that may be used to provide the ability to reference and classify elements and namespaces.
+
+This document includes the elements as defined in the Refract specification.
+
+## Referencing and Identifying Elements
+
+These elements provide the means to identify elements with unique or reused qualifiers.
+
+### Ref (Element)
+
+The `ref` element provides a way to reference elements and namespaces with a URI. The context the `ref` element is found in determines what exactly it references. Additionally, the referenced
+
+#### Properties
+
+- `element`: ref (string, required)
+- `content` (string, required) - The URI of the referenced item
+
+### ID (Element)
+
+The `id` element provides a document-wide unique identifier. It MUST be unique for the entire document in which is found. When used as an attribute for another element, it defines the unique ID for that element, and MAY be used in the fragment identifier to link to that element.
+
+#### Properties
+
+- `element`: id (string, required)
+- `content` (string, required) - The unique identifier for an element
+
+#### Example
+
+This example shows the `id` element used as an attribute to provide a unique identifier.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "id": "my-foo"
+  },
+  "content": "bar"
+}
+```
+
+In this example, the `foo` element has an `id` of `my-foo`, which can be used to reference this specific instance of the `foo` element.
+
+### Roles (Element)
```

Comment:
Further, given there is no concept with roles associated with referencing here, should this be in the base spec. Or if there is an aspect of referencing, needs an example.